sources/views/contacts.js

```diff
@@ -1,81 +1,87 @@
 			scrollX:false,
-			select:true
-		};
-		var contacts_form = {
-			view:"form",
-			id:"contacts_form",
-			width:350,
-			elements:[
-				{
-					margin:10,
-					rows:[
-						{template:"EDIT CONTACTS", type:"section"},
-						{view:"text", label:"Name", name:"Name", invalidMessage:"Enter correct Name"},
-						{view:"text", label:"Email", name:"Email", invalidMessage:"Enter correct Email"},
-						{margin:20, cols:[
-							{view:"button", value:"Save", css:"webix_primary", click:() => this.saveHandler("contacts_form")},
-							{view:"button", value:"Clear", css:"webix_secondary", click:() => this.clearHandler("contacts_form")}
-						]}
-					]
-				},{}
-			],
-			rules:{
-				Name:function(value){
-					return textValidation(value,20);
-				},
-				Email:webix.rules.isEmail
-			}
+			select:true,
+			on:{ onAfterSelect:(id) => this.selectContact(id)},
+			onClick:{ remove_list_item_btn:(e,id) => this.removeContact(id) }
 		};
+
 		return {
 			cols:[
-				contacts_list,
-				contacts_form
+				{rows:[
+					contacts_list,
+					add_new_button
+				]},
+				FormView
 			]
 		};
 	}
-	init(){
-		webix.$$("contacts_list").parse(contacts);
+
+	addContact(){
+		const _ = this.app.getService("locale")._;
+		webix.message({text:_("New element was added"), expire:350});
+		const new_contact = { Name:"Name", Email:"Email", Status:0, Country:0};
+		contacts.add(new_contact);
+
+		const new_id = contacts.getLastId();
+		this.getRoot().queryView("list").select(new_id);
 	}
-	clearHandler(form_id){
-		var form_collection = webix.$$(form_id);
+	removeContact(id){
+		const _ = this.app.getService("locale")._;
 		webix.confirm({
-			title:"Form would be cleared",
-			text:"Do you still want to continue?",
+			title:_("Contact data would be deleted"),
+			text:_("Do you still want to continue?"),
 			type:"confirm-warning"
-		}).then(
-			function(){
-				form_collection.clear();
-				form_collection.clearValidation();
-			},
-			function(){
-				webix.message("Rejected");
-			});
-	}
-	saveHandler(form_id){
-		var form_collection = webix.$$(form_id);
-		if(form_collection.validate()){
+		}).then(() => {
 			webix.message({
-				text:"Data has edited successfully",
-				type:"success",
-				expire:3000
-			});
-			form_collection.clear();
-		}else{
-			webix.message({
-				text:"Please, enter the correct data in the fields of the form",
-				type:"error",
-				expire:3000
+				text:_("Element was deleted"),
+				type:"info"
 			});
+
+			contacts.remove(id);
+			this.show("contacts");
+			return false;
+		},
+		function(){
+			webix.message(_("Rejected"));
+		});
+	}
+	selectContact(id){
+		this.setParam("id", id, true);
+	}
+
+	init(view){
+		const list = view.queryView("list");
+		list.parse(contacts);
+	}
+
+	urlChange(view){
+		const list = view.queryView("list");
+		const url_id = this.getParam("id");
+		const first_id = contacts.getFirstId();
+
+		if(url_id && contacts.exists(url_id)){
+			list.select(url_id);
+		}else if(first_id){
```

Comment:
small optimization: you do not need to define a variable (first_id) that you will probably not use 

the code may look like 
```
const id = this.getParam("id") || contacts.getFirstId();
if(contacts.exists(id))
   list.select(id);
```

Reply:
It is a really cool optimization

Reply:
ok. it looks better